src/csv.jl

```diff
@@ -399,25 +400,25 @@ function _csvread_internal(str::AbstractString, delim=',';
     cols, canonnames, parsers, finalrows
 end
 
-function promote_field(failed_str, field, col, err, nastrings, stringtype, opts)
+function promote_field(failed_str, field, col, err, nastrings, stringtype, stringarraytype, opts)
     newtoken = guesstoken(failed_str, opts, field.inner, nastrings)
     if newtoken == field.inner
         # no need to change
         return field, col
     end
     newcol = try
-        promote_column(col,  err.rowno-1, fieldtype(newtoken), stringtype)
+        promote_column(col,  err.rowno-1, fieldtype(newtoken), stringtype, stringarraytype)
     catch err2
         Base.showerror(stderr, err2)
         rethrow(err)
     end
     swapinner(field, newtoken), newcol
 end
 
-function promote_column(col, rowno, T, stringtype, inner=false)
+function promote_column(col, rowno, T, stringtype, stringarraytype, inner=false)
     if typeof(col) <: Array{Missing}
         if T <: StringLike
-            arr = StringVector{stringtype}(length(col))
+            arr = stringarraytype{stringtype,1}(undef, length(col))
```

Comment:
does this method work for `StringVector`?

Reply:
It should with https://github.com/JuliaData/WeakRefStrings.jl/pull/48 merged.